src/abstract/ZoweTreeProvider.ts

```diff
@@ -141,39 +141,49 @@ export class ZoweTreeProvider {
         return undefined;
     }
 
-    public async editSession(node: IZoweTreeNode) {
+    public async editSession(node: IZoweTreeNode, zoweFileProvider: IZoweTree<IZoweNodeType>) {
         const profile = node.getProfile();
         const profileName = node.getProfileName();
-        // Check what happens is inactive
-        await Profiles.getInstance().getProfileSetting(profile);
-        const EditSession = await Profiles.getInstance().editSession(profile, profileName);
-        if (EditSession) {
-            node.getProfile().profile = EditSession as IProfile;
-            await setProfile(node, EditSession as IProfile);
-            await setSession(node, EditSession as ISession);
-            this.refresh();
-        }
-        try {
-            // refresh profilesForValidation to check the profile status again
-            Profiles.getInstance().profilesForValidation.forEach((checkProfile, index) => {
-                if (index === 0) {
-                    Profiles.getInstance().profilesForValidation = [];
-                }
-                if (checkProfile.name === profileName) {
-                    Profiles.getInstance().profilesForValidation.splice(index,1);
+
+        // Check what happens if inactive
+        const newProfile = await Profiles.getInstance().editSession(profile, profileName);
+
+        if (newProfile) {
+            // Get the active session
+            let EditSession;
+            try {
+                EditSession = await ZoweExplorerApiRegister.getMvsApi(node.getProfile()).getSession(null, false);
```

Comment:
I'm not sure if we want to use `ZoweExplorerApiRegister.getMvsApi()` here outside of the MVS context, as not all extenders might actually register the MVS API. For example (@phaumer correct me if I'm wrong), but I believe the zFTP Zowe Explorer extension actually only registers the USS API as of now (although we will hopefully add MVS for zFTP this PI).

Reply:
@katelynienaber @JillieBeanSim Why do we need getSession again? This was added in Edit Session and Check Current Session.

Reply:
@jellypuno Because once you have edited the session, we have to add the new (edited) information to the node's session & profile. Otherwise Explorer will use the old info on the node, and it will be like you never edited it at all.